File: Programmers/코딩테스트_입문/GangHyeok/Day25/연속된 수의 합.java
class Solution {
    public int[] solution(int num, int total) {
        int[] answer = new int[num];
        
        int a, idx;
        a = total / num;
        
        if(num%2 == 1){
            idx = num/2;
        }
        else{
            idx = num/2-1;
        }
        
        for(int i=0;i<num;i++){
            answer[i] = a;
        }
        
        for(int i=0;i+idx<num;i++){
            answer[i+idx]+=i;
        }
        
        for(int i=0;idx-i>=0;i++){
            answer[idx-i]-=i;
        }
        
        return answer;
    }
}

File: Programmers/코딩테스트_입문/GangHyeok/Day25/연속된 수의 합.js
function solution(num, total) {
  var answer = [];
  
  // 연속된 수 => 1~num까지의 합을 구하고 주어진 total에 빼기
  // total과 num까지 합과의 차를 num으로 나눠서 배열 각 요소에 골고루 나눠주기
  for(let i =1;i<=num;i++){
      answer[i-1] = i + (total - num*(num+1)/2)/num;
  }  
  return answer;
}